pkg/autoscaler/autoscaler.go

```diff
@@ -343,6 +447,106 @@ func (a *Autoscaler) Scale(ctx context.Context, now time.Time) (int32, bool) {
 	}
 
 	a.reporter.Report(DesiredPodCountM, float64(desiredPodCount))
+	logger.Infof("desiredPodCount = %v", desiredPodCount)
+	return desiredPodCount, true
+}
+
+func (a *Autoscaler) scaleBasedOnSamples(ctx context.Context, now time.Time) (int32, bool) {
```

Comment:
Again, I'm not sure why you have a special and separate function for scaling based on samples.